tools/read-conf.py

```diff
@@ -7,58 +7,115 @@
 import sys
 
 
-def getlist(conf_section, key, fallback=None):
-    string = conf_section.get(key, fallback=fallback)
-    if string:
-        return re.split(r"[, ]+", string)
-    else:
-        return fallback
+def get_string(conf, key, fallback=None):
+    return conf.get(key, fallback=fallback)
 
-def check_config(conf, result):
-    # Some error checking:
-    for section in conf.sections():
-        if section not in ['APY', 'REPLACEMENTS']:
-            raise configparser.Error("\nUnknown section [%s]" % (section,))
+def get_string_array(conf, key, fallback=None):
+    string = get_string(conf, key)
+    return fallback if string is None else re.split(r'[, ]+', string)
 
-    apy_diff = set(k.lower() for k in conf['APY'].keys()) - set(k.lower() for k in result.keys())
-    if apy_diff:
-        raise configparser.Error("\nUnknown key(s) in section [APY]: %s" % (apy_diff,))
+def get_bool(conf, key, fallback=None):
+    return conf.getboolean(key, fallback=fallback)
 
-    return True
+def get_int(conf, key, fallback=None):
+    return conf.getint(key, fallback=fallback)
 
-def load_conf(filename):
-    conf = configparser.ConfigParser()
-    with open(filename, 'r') as f:
-        conf.read_file(f)
-    conf_APY = conf['APY']
-    result = {
-        'HTML_URL'                       : conf_APY.get('HTML_URL', fallback="http://www.apertium.org"),
-        'APY_URL'                        : conf_APY.get('APY_URL', fallback="http://apy.projectjj.com"),
+def get_int_array(conf, key, fallback=None):
+    array = get_string_array(conf, key)
+    return fallback if array is None else [int(x) for x in array]
+
+
+# Constant descriptors
+constants = {
+    # section name
+    'APY': {
+        # CONSTANT_NAME: [parser]
+        # CONSTANT_NAME: [parser, fallback]
```

Comment:
This should not be an list, a tuple is a far far better choice.

Reply:
Sure